src/map/packets/quest_mission_log.h

```diff
@@ -53,38 +98,60 @@
 #define ADO_COMPLETE		0xF8
 #define COA_COMPLETE		0x0108
 
-//Mission Log Types
 #define MISS_COMPLETE		0xD0
 #define MISS_CURRENT		0xFFFF
 
-// MISS_CURRENT	0xFF исключая Add-on Scenarios
+#define EXP_MISS_COMPLETE   0xD8
+#define EXP_MISS_CURRENT    0x80
 
-#define EXP_MISS_COMPLETE               0xD8
-#define EXP_MISS_CURRENT                0x80
+#define CMPGN_MISS_UN       0x30
+#define CMPGN_MISS_DEUX     0x38
 
-#define CAMPAIGN_MISSION_UN             0x30
-#define CAMPAIGN_MISSION_DEUX           0x38
+// Status Types
+#define STATUS_QUEST_CURR   0x01
+#define STATUS_QUEST_COMP   0x02
+#define STATUS_MISS_CURR    0x03
+#define STATUS_MISS_COMP    0x04
 
-// Mission Logs
-#define MISSION_SANDORIA                0x0B
-#define MISSION_BASTOK                  0x0C
-#define MISSION_WINDURST                0x0D
-#define MISSION_ZILART                  0x0E
-#define MISSION_TOAU                    0x0F
-#define MISSION_WOTG                    0x10
-#define MISSION_COP                     0x11
-
-#define MISSION_ASSAULT                 0x12
-
-#define MISSION_CAMPAIGN                0x13
-#define MISSION_CAMPAIGN2               0x14
-
-#define MISSION_ACP                     0x15
-#define MISSION_AMK		                0x16
-#define MISSION_ASA		                0x17
-
-#define MISSION_SOA					    0x18
-#define MISSION_ROV                     0x19
+// Log Types
+const int LOG_TYPES[34][7] =
```

Comment:
By invalid do you mean references called in the wrong context linking to invalid data (Tavnazia, Fame: -1)? All valid references point to the correct values in a character's set of mission logs.

Regardless of which C/Lua side the table is on, we'll still need some sort of indication (-1) that a subject was called in an inappropriate context. Transitioning to a key-based lookup (in Lua, prior to sending it to the Lunar binding somehow) with individual tables (mission only, quest only, etc) and flagging an error if the key wasn't found would avoid explicitly listing invalid references. But that requires mass-replacing calls in all existing scripts (from Lunar-bound player:completeQuest to something like a Lua-intermediary player_util:completeQuest), which is what I was attempting to avoid.

The replacing itself wouldn't be an issue, but all non-vanilla-DSP scripts would break and probably result in several unhappy server operators. Not that I personally mind; just trying to avoid making waves.

I will also admit a personal preference to referencing/verification on C-side as opposed to script side because that's less likely to get screwed up. The lookup occurring on the back-end means it'll always result in the correct value. But if it's on Lua's end, someone could use the old (current) player:completeQuest instead of the new one (maybe they're using an old guide/documentation, or simply didn't get the memo) to result in the server pulling/saving in the wrong log location. But I fully acknowledge that's just personal preference based on a latent attitude of distrust.

Is mass-replacing the existing calls an avenue you'd rather pursue?